payjoin/src/into_url.rs

```diff
@@ -0,0 +1,109 @@
+use url::{ParseError, Url};
+
+#[derive(Debug)]
+pub enum Error {
+    BadScheme,
+    ParseError(ParseError),
+}
+
+impl std::fmt::Display for Error {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Error::BadScheme => write!(f, "URL scheme is not allowed"),
+            Error::ParseError(e) => write!(f, "{}", e),
+        }
+    }
+}
```

Comment:
A neat trick I saw in rust-bitcoin to avoid repetition:

```suggestion
impl std::fmt::Display for Error {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        use Error::*;
        
        match self {
           BadScheme => write!(f, "URL scheme is not allowed"),
           ParseError(e) => write!(f, "{}", e),
        }
    }
}
```